### PR TITLE
Lower gas usage for creating a Multisig Wallet

### PR DESCRIPTION
**Pull request by ngotchac**
This introduces a new Multisig Wallet that only uses about 500k gas to deploy (compared to ~1.8 to 2M gas for the previous version).

This uses a Multisig Wallet Library that has to be deployed to the network (if not, it fallbacks to the old one).  
One has been deployed to Ropsten already. To deploy a new one:

- Get the bytecode in `js/src/contracts/code/wallet.js`
- Deploy it (if using the UI, ABI set to `[]` is fine)
- Register the contract address in the Registry with the key specified in `js/src/contracts/code/wallet.js` : `walletLibrary`

For new versions, when the Wallet is compiled, modify in the bytecode the references to `cafecafe...` with `__WalletLibrary___________`

The new contract is in `js/src/contracts/snippets/enhanced-wallet.sol`

### CODE REVIEW

**Comment by jacogr** [comment]
We should be doing the library deploys using the Parity keys. So when you are happy, let me know and I can deploy & register using the required keys.

I'm assuming there is no issues across multiple networks, e.g. Ropsten & Homestead from a UI & knowing where to find stuff perspective?

**Comment by ngotchac** [comment]
Okay sure. I registered it on Ropsten, so you can test. When everything is fine, I'll transfer the ownership of the registered key on Ropsten.
Nothing network related on the UI.

File: js/src/modals/CreateWallet/createWalletStore.js [hunk 1]
```diff
@@ -17,9 +17,11 @@
 import { observable, computed, action, transaction } from 'mobx';
 
 import Contract from '~/api/contract';
+import Contracts from '~/contracts';
 import { ERROR_CODES } from '~/api/transport/error';
 import { wallet as walletAbi } from '~/contracts/abi';
 import { wallet as walletCode } from '~/contracts/code';
+import { walletLibraryRegKey, fullWalletCode } from '~/contracts/code/wallet';
```
Comment (ngotchac) [inline js/src/modals/CreateWallet/createWalletStore.js]: Fixed

**Comment by gavofyork** [comment]
there's ethcore/contracts repo for any contract stuff we do...

**Pull request merged**